### PR TITLE
[REF] Move remaining locale functions to _config.localization

### PR DESCRIPTION
**Pull request by jbrockmendel**
There are a a couple locale-centric functions left in `tm` that have not yet been moved because they depend on `pd.options.display.encoding`.  This PR implements a minimal `_config.display` that allows us to consolidate the locale functions.  `_config.display` also defines `display.date_dayfirst` and `display.date_yearfirst`, since those are used by `tslibs.parsing`.  So this completes the process of removing `tslib`'s dependency on `core`.

The existing `get_locales` function has a `if PY3` check that is no longer necessary, so this removes the PY2 branch of that.

### CODE REVIEW

File: pandas/_config/display.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+
+def detect_console_encoding():
+    """
+    Try to find the most capable encoding supported by the console.
+    slightly modified from the way IPython handles the same issue.
+    """
+    global _initial_defencoding
+
+    encoding = None
```
Comment (jreback) [inline pandas/_config/display.py]: shouldn’t u check the global first? (if it’s set)

File: pandas/_config/display.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    these are generally strings meant to be displayed on the console.
+"""
+
+pc_date_dayfirst_doc = """
+: boolean
+    When True, prints and parses dates with the day first, eg 20/01/2005
+"""
+
+pc_date_yearfirst_doc = """
+: boolean
```
Comment (jreback) [inline pandas/_config/display.py]: these are pretty orthogonal right ?

File: pandas/_config/display.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_config/display.py]: The comment near the bottom suggests this is kept for debugging purposes related to matplotlib

File: pandas/_config/display.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_config/display.py]: Yes.  Handling dayfirst and yearfirst finishes breaking the reliance of tslibs on core.

Comment (jreback) [reply pandas/_config/display.py]: right but they shouldn't be in this file. not problem separateing them out, but not here

Comment (jbrockmendel) [reply pandas/_config/display.py]: they're under the `display` namespace within `options`.  where would you put this?  could go directly at the bottom of `__init__` i guess

Comment (jreback) [reply pandas/_config/display.py]: i would put them in another file under _config

Comment (jbrockmendel) [reply pandas/_config/display.py]: done + green

**Comment by jreback** [comment]
thanks @jbrockmendel
